### PR TITLE
Format all Markdown files using Prettier

### PR DESCRIPTION
**Pull request by lipis**
Prettier is now supporting Markdown since v1.8.0 and also formats the code inside the fences.

https://prettier.io

If something is actually breaking or it's wrong, please let me know as we can add a few ignore blocks in that case.

### CODE REVIEW

File: docs/api/locales.md [hunk 1]
```diff
@@ -2,141 +2,140 @@
...
+| ur            | Urdu                    |
+| uz            | Uzbek                   |
+| vi            | Vietnamese              |
+| xh            | Xhosa                   |
+| yi            | Yiddish                 |
+| yo            | Yoruba                  |
+| zh            | Chinese                 |
+| zh-CN         | Chinese (Simplified)    |
+| zh-TW         | Chinese (Traditional)   |
+| zu            | Zulu                    |
```
Comment (lipis) [inline docs/api/locales.md]: This is just beautiful :D

File: package.json [hunk 2]
```diff
@@ -33,6 +33,13 @@
       "browser": true
     }
   },
+  "prettier": {
+    "bracketSpacing": false,
+    "printWidth": 80,
+    "proseWrap": false,
+    "semi": false,
+    "singleQuote": true
```
Comment (lipis) [inline package.json]: These are the options that are used for Markdown and the code inside the fences. For more options check the full list: https://prettier.io/docs/en/options.html

**Comment by codebytere** [comment]
Hi! We appreciate the contribution here but we have our own markdown linter and don’t use prettier as a format standard within our js. This PR unfortunately breaks our markdown parser and so at this time this isn’t something we think we can integrate.

**Comment by jkleinsc** [comment]
Closing due to incompatibility with our markdown parser.  Feel free to reopen if there is a way to make it work with our markdown parser.

**Comment by lipis** [comment]
Are you guys talking about the [`standard-markdown`](https://github.com/zeke/standard-markdown).. this from their README:

> This module works just like standard, but instead of linting javascript files, it lints GitHub-Flavored js and javascript code blocks inside markdown files.

Which means that it is not actually formatting the markdown files but only the code inside of them.

What Prettier does on the other hand.. is actually normalising all the files and keeps the consistency among all of them. On top of that it also out of the box formats the code inside as well and I simply modified the options to match the `standard` format.. If that is not good enough (you can check the diffs).. I can update this and parse the files with `standard-markdown` to update the code in the desired format.

So basically prettier does what standard-markdown is doing plus formatting the actual file.

**Comment by lipis** [comment]
I'm suggesting reconsider and of course I'm fully aware that the standard formatting is being used, and this PR is not affecting anything more than the Markdown files..

**Comment by jkleinsc** [comment]
@lipis we do several things with our markdown files:
1. We lint them against `standard-markdown`
2. We generate an api file (electron-api.json)
3. We generate typescript definitions

You can test all three of these by running `npm run lint-docs`.   If that passes your change could be considered, but right now they are not passing.

**Comment by lipis** [comment]
OKey.. I'll take another look.. :)

**Comment by lipis** [comment]
So I'm getting these errors and there is no such option in Prettier.. is that really so important.. or this can be skipped?

```
...

> electron@1.8.2-beta.2 lint-js-in-markdown /Users/lipis/work/lipis/electron
> standard-markdown docs

   api/desktop-capturer.md
         38:22:    Missing space before function parentheses.
         44:21:    Missing space before function parentheses.

   api/menu.md
         148:14:   Missing space before function parentheses.

   api/protocol.md
         197:22:   Missing space before function parentheses.

   api/sandbox-option.md
         90:26:    Missing space before function parentheses.

   api/session.md
         362:25:   Missing space before function parentheses.
         366:13:   Missing space before function parentheses.
         370:13:   Missing space before function parentheses.

   api/web-frame.md
         77:13:    Missing space before function parentheses.

   development/build-system-overview.md
         68:39:    Missing space before function parentheses.

   tutorial/accessibility.md
         16:46:    Missing space before function parentheses.

   tutorial/desktop-environment-integration.md
         63:10:    Missing space before function parentheses.
         144:10:   Missing space before function parentheses.
         152:10:   Missing space before function parentheses.

   tutorial/planned-breaking-changes.md
         122:38:   Missing space before function parentheses.
         126:38:   Missing space before function parentheses.

   tutorial/quick-start.md
         61:22:    Missing space before function parentheses.

   tutorial/using-selenium-and-webdriver.md
         26:17:    Missing space before function parentheses.
         30:17:    Missing space before function parentheses.
         34:17:    Missing space before function parentheses.
         38:17:    Missing space before function parentheses.
         42:18:    Missing space before function parentheses.
         46:17:    Missing space before function parentheses.

There are 23 errors in "docs"
```

**Comment by jkleinsc** [comment]
@lipis yup... they are important as they are needed in the generation of the api file and typescript definitions.

**Comment by lipis** [comment]
Let's have this one open for a while.. maybe we'll get somewhere #11156 /cc @ikatyang
